named_arrays/_functions/functions.py

```diff
@@ -66,6 +108,21 @@ def value(self) -> FunctionArray:
             outputs=self.outputs.value,
         )
 
+    @property
+    def broadcasted(self) -> FunctionArray:
+
+        broadcasted_shape_outputs = self.shape
+        broadcasted_shape_inputs = self.shape
+        for key in broadcasted_shape_outputs:
+            axes_vertex = self.axes_vertex
+            if key in axes_vertex:
+                broadcasted_shape_inputs[key] = self.inputs.shape[key]
+
+        return FunctionArray(
+            self.inputs.broadcast_to(broadcasted_shape_inputs),
+            self.outputs.broadcast_to(broadcasted_shape_outputs)
+       )
```

Comment:
We shouldn't assume the type of the output, it could be an IRIS function array or something

```suggestion
return dataclasses.replace(
    self,
    inputs=self.inputs.broadcast_to(broadcasted_shape_inputs),
    outputs=self.outputs.broadcast_to(broadcasted_shape_outputs),
)
```